### DISCUSSION TITLE
IPv6 disablement isn't working

### DISCUSSION THREAD

**Discussion by ghost** [DNSCrypt/dnscrypt-proxy#2580] [Q&A]
Rather than posting this as an issue, I am posting this as a discussion to make sure I am not misunderstanding DNSCrypt-Proxy configuration on Windows 11 OS. I want to disable IPv6 servers and resolving to any IPv6 addresses, but it doesn't work. Here are the 2 configuration changes that are supposed to disable IPv6, but I receive IPv6 responses regardless:
ipv6_servers = false
block_ipv6 = true

I also tried disabling HTTP3 DoH, but that also didn't prevent IPv6 responses.

**Comment by ghost** [comment]
Do you mean that you still get the correct IPv6 address as a response?
Because what `block_ipv6 = true` does is giving an empty synthetic response, instead of the actual one.

**Reply by ghost** [reply]
I test via BrowserLeaks.com DNS leak test (https://browserleaks.com/ip). When I use my IPv4-only VPN DNS server, which is IPv4-only, then I only get IPv4 responses in BrowserLeaks DNS leak test. 

When I use YogaDNS (which uses DNSCrypt-Proxy library and same type of configuration file) with block_ipv6=true and ip6_servers=false, then I don't get IPv6 responses in BrowserLeaks DNS leak test. 

When I use actual DNSCrypt-Proxy from this repository (with block_ipv6=true and ip6_servers=false), BrowserLeaks DNS leak test shows both - IPv4 and IPv6 servers.

**Reply by ghost** [reply]
Did you use the same DNS server (one which also has a IPv6, even if it's not used now) in all 3 cases?

It seems to me BrowserLeaks is just listing the IPs of whatever DNS servers one is using. Obviously we cannot prevent a 3rd party from resolving IPv6, too.

**Comment by ghost** [comment]
You mean proxied responses? I disabled ODoH and didn't enable Anon Relays. I just tested NSLOOKUP command and it didn't return any IPv6 addresses... I guess BrowserLeaks does something... else?

**Comment by welwood08** [comment]
The DNS leak test at https://browserleaks.com/dns does not tell you about any DNS responses you receive, it tells you which DNS resolvers connected to the BrowserLeaks service during the test. If you see IPv6 addresses on that test, it means a resolver with that address requested information on your behalf. The tool should be used to identify if for example your ISP resolver requests information when you expect only VPN resolvers, not to test if locally disabling IPv6 had any effect.

**Comment by ghost** [comment]
Yes, I tested other DNS Leak Test sites and BrowserLeaks is the only one reporting IPv6 results. It does something other DNS Leak Test sites don't. Thank you for clarification.
